pandas/tests/groupby/test_groupby.py

```diff
@@ -5777,6 +5769,81 @@ def test_agg_over_numpy_arrays(self):
 
         assert_frame_equal(result, expected)
 
+    def test_cummin_cummax(self):
+        # GH 15048
+        # Test with different dtypes
+        for dtype in [np.int32, np.float32, np.float64]:
+            df = pd.DataFrame({'A': [1, 1, 1, 2, 2, 2],
+                               'B': range(3, 9)}).astype(dtype)
+            result = df.groupby('A').cummin()
+            expected = pd.DataFrame({'B': [3, 3, 3, 6, 6, 6]}).astype(dtype)
+            tm.assert_frame_equal(result, expected)
+            expected = df.groupby('A').B.apply(lambda x: x.cummin()).to_frame()
+            tm.assert_frame_equal(result, expected)
+            # Test min value for dtype
+            try:
+                min_val = np.iinfo(dtype).min
+            except ValueError:
+                min_val = np.finfo(dtype).min
+            df.loc[[1, 4], 'B'] = min_val
+            result = df.groupby('A').cummin()
+            expected = pd.DataFrame({'B': [3, min_val, min_val,
+                                           6, min_val, min_val]}).astype(dtype)
+            tm.assert_frame_equal(result, expected)
+            expected = df.groupby('A').B.apply(lambda x: x.cummin()).to_frame()
+            tm.assert_frame_equal(result, expected)
+
+            df = pd.DataFrame({'A': [1, 1, 1, 2, 2, 2],
+                               'B': range(8, 2, -1)}).astype(dtype)
+            result = df.groupby('A').cummax()
+            expected = pd.DataFrame({'B': [8, 8, 8, 5, 5, 5]}).astype(dtype)
+            tm.assert_frame_equal(result, expected)
+            expected = df.groupby('A').B.apply(lambda x: x.cummax()).to_frame()
+            tm.assert_frame_equal(result, expected)
+            # Test max value for dtype
+            try:
+                max_val = np.iinfo(dtype).max
+            except ValueError:
+                max_val = np.finfo(dtype).max
+            df.loc[[1, 4], 'B'] = max_val
+            result = df.groupby('A').cummax()
+            expected = pd.DataFrame({'B': [8, max_val, max_val,
+                                           5, max_val, max_val]}).astype(dtype)
+            tm.assert_frame_equal(result, expected)
+            expected = df.groupby('A').B.apply(lambda x: x.cummax()).to_frame()
+            tm.assert_frame_equal(result, expected)
+
+        # Some entries in column are nan
```

Comment:
so this needs to work for float types (32 and 64). as well as datetime64/timedelta64 types (which use NaT)